backend/src/akvo/lumen/lib/visualisation/map_config.clj

```diff
@@ -111,12 +113,16 @@
       (str/replace #"\n" " ")
       (str/replace #" +" " ")))
 
-;; Should convert hex string to hue for hsl color. Use string matching as proof of concept for now
-(defn color-to-hue [s]
-  (case s
-    "#FF0000" "0"
-    "#00FF00" "120"
-    "#0000FF" "240"))
+(defn color-to-hue
+  "Should convert hex string to hue for hsl color."
+  [hex-string]
+  (try
+    (let [c (Color/decode hex-string)
+          hue (->> (Color/RGBtoHSB (.getRed c) (.getGreen c) (.getBlue c) (float-array 3))
+                   (into [])
```

Comment:
`(first (float-array 3)) => 0.0` why the need to convert to vector with `(into [])` ?

Reply:
you're right!